apps/music/test/unit/metadata/utils.js

```diff
@@ -29,20 +29,34 @@ function parseMetadata(filename) {
   var songBlob;
   return _fetch(filename).then(function(data) {
     songBlob = new Blob([data]);
+    songBlob.name = filename;
     return AudioMetadata.parse(songBlob);
   }).then(function(metadata) {
     if (metadata.picture) {
       return new Promise(function(resolve, reject) {
         var reader = new FileReader();
-        var coverBlob = metadata.picture.blob || songBlob.slice(
-          metadata.picture.start, metadata.picture.end, metadata.picture.type
-        );
+
+        var coverBlob = null;
+        if (metadata.picture.filename) {
+          var storage = navigator.getDeviceStorage('pictures');
+          var request = storage.get(metadata.picture.filename);
+          request.onsuccess = function() {
+            coverBlob = this.result;
+          };
+        } else {
+          coverBlob = metadata.picture.blob || songBlob.slice(
+            metadata.picture.start, metadata.picture.end, metadata.picture.type
+          );
+        }
```

Comment:
This whole block shouldn't be necessary; it's not needed for the existing unsynced ID3 art tests, so the same should apply here. If the tests fail without this change, that suggests there's a bug, since the existing code tests the flow for song files coming from an "open" activity (where the files have no filename).

(I plan on adding tests for the "regular file" state later, but I don't think that needs to be done here.)